### PR TITLE
ref: move TestNSNotificationCenterWrapper.swift from SentryTests to SentryTestUtils

### PR DESCRIPTION
**Pull request by armcknight**
Finally figured out the necessary changes to get this in so I can use it in the continuous profiling tests.

I still had to reimplement SentryNotificationCenterWrapperTests from objc to swift because I couldn't figure out how to get the test subclass to appear in SentryTestUtils-Swift.h (which isn't actually used anywhere as a thing, as yet) so it could be imported into the objc test implementation.

#skip-changelog

### CODE REVIEW

File: Sentry.xcodeproj/xcshareddata/xcschemes/Sentry.xcscheme [hunk 1]
```diff
@@ -24,8 +24,8 @@
    </BuildAction>
    <TestAction
       buildConfiguration = "Test"
-      selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
-      selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
+      selectedDebuggerIdentifier = ""
+      selectedLauncherIdentifier = "Xcode.IDEFoundation.Launcher.PosixSpawn"
```
Comment (philipphofmann) [inline Sentry.xcodeproj/xcshareddata/xcschemes/Sentry.xcscheme:28]: `m`: Did you change this on purpose?

**Review by philipphofmann — APPROVED** [review]
Thanks for this 💯 . Our mix of Swift and ObjC is sometimes a pain, but we'll figure it out.

File: Sentry.xcodeproj/xcshareddata/xcschemes/Sentry.xcscheme (hunk 1 above)
Comment (armcknight) [reply Sentry.xcodeproj/xcshareddata/xcschemes/Sentry.xcscheme:28]: nope, thanks for mentioning this!

**Comment by armcknight** [comment]
@philipphofmann thanks for reviewing... i had made another change while flying home saturday, and forgot to push the commit. check out https://github.com/getsentry/sentry-cocoa/pull/3968/commits/683ffb38bf98ddab789b522bfbbb081e9ca26c48 when you get a chance, it helps being able to fire a notification in the mock

**Review by philipphofmann — APPROVED** [review]
LGTM

**Pull request merged**